src/realm/table.cpp

```diff
@@ -3530,8 +3560,8 @@ void Table::typed_print(std::string prefix, ref_type ref) const
 
 StringInterner* Table::get_string_interner(ColKey::Idx idx) const
 {
-    REALM_ASSERT(idx.val < m_string_interners.size());
-    auto interner = m_string_interners[idx.val].get();
-    REALM_ASSERT(interner);
-    return interner;
+    const auto index = idx.val;
```

Comment:
I don't think the assertion should be removed. If you can't get an interner, it is a sign that you got an error in the call. There is a problem in `StringNodeFulltext::table_changed()` where the base class should not be called.